grafana/dashboards.sh

```diff
@@ -36,82 +61,346 @@ DASHBOARDS_DIRECTORY: $DASHBOARDS_DIRECTORY
 }
 
 backup() {
+	local dashboard_info_json
+	local dashboard_uid
+	local dashboard_title
+	local initial_folder_uid
+	local grafana_path_array_str
+	local -a grafana_path_array # Declare as array
 	local dashboard_json
+	local target_dir
+	local filename
 
-	for dashboard in $(list_dashboards); do
-		dashboard_json=$(get_dashboard "$dashboard")
+	echo "Starting dashboard backup process (scoped to 'TeslaMate' folder and its subfolders)..."
+	mkdir -p "$DASHBOARDS_DIRECTORY" # Ensure base directory exists
 
-		if [[ -z $dashboard_json ]]; then
-			echo "ERROR:
-  Couldn't retrieve dashboard $dashboard.
-      "
-			exit 1
+	list_dashboards | while IFS= read -r dashboard_info_json; do
+		dashboard_uid=$(echo "$dashboard_info_json" | jq -r '.uid')
+		dashboard_title=$(echo "$dashboard_info_json" | jq -r '.title')
```

Comment:
Maybe I am being excessively paranoid. But the lack of double quotes around many of these variables in a shell script makes me nervous.

```sh
dashboard_title="$(echo "$dashboard_info_json" | jq -r '.title')"
```

Otherwise if the title happens to have spaces it could cause problems.

Reply:
Looking at `dutch-tax.json`: it has the title "Drives - Dutch tax". This results in:

```
BACKED UP: ./grafana/dashboards/reports/drives---dutch-tax.json (UID: lBIoQIggk)
```

So it's not the same filename as before (that is the case for two dashboards: this one and `database-information.json` instead of `database-info.json`), but it doesn't look like there's a problem with spaces, does it?

Reply:
Yes, might be OK. Still makes me somewhat nervous though.

Reply:
No problem, I've added quotes.